webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -1026,6 +1027,10 @@ func (v validator) validateUpdatesWhenPoweredOn(ctx *pkgctx.WebhookRequestContex
 		allErrs = append(allErrs, field.Forbidden(specPath.Child("bootstrap"), updatesNotAllowedWhenPowerOn))
 	}
 
+	if !equality.Semantic.DeepEqual(vm.Spec.GuestID, oldVM.Spec.GuestID) {
```

Comment:
Can we get by with `!=` here instead?

Reply:
We also have a static list of valid guest IDs that could be used for validation: https://github.com/vmware/govmomi/blob/main/vim25/types/enum.go#L13439

Reply:
Fixed to use `!=`.
@dougm This field used to be validated by the above enum list in the API, but we switched to a string to avoid blocking VM Service VM creation/update in case a newer guestID is used, which hasn't yet been updated in the govmomi repo.